numba/cuda/tests/cudapy/test_sm.py

```diff
@@ -42,8 +42,22 @@ def test_issue_1051_shared_size_broken_2d(self):
         self._check_shared_array_size((2, 3), 6)
 
     def test_issue_1051_shared_size_broken_3d(self):
+
         self._check_shared_array_size((2, 3, 4), 24)
 
+    def _check_shared_array_size_fp16(self, shape, expected):
+        @cuda.jit
+        def s(a):
+            arr = cuda.shared.array(shape, dtype=types.float16)
```

Comment:
Do you have a test for the case where `np.float16` is used?

Reply:
@gmarkall test case added.